src/SeleniumLibrary/keywords/element.py

```diff
@@ -164,21 +164,47 @@ def page_should_contain_element(self, locator, message='', loglevel='INFO'):
         self.assert_page_contains(locator, message=message, loglevel=loglevel)
 
     @keyword
+    def get_matching_locator_count(self, locator):
+        """Returns number of elements matching ``locator``
+
+        If you wish to assert the number of matching elements, use
+        `Locator Should Match X Times`.
+
+        Examples assumes that locator matches to three elements
+        | ${count} =                 | Get Matching Locator Count | xpath: //*[@name="div_name"] |
+        | Should Be Equal As Numbers | ${count}                   | 3                            |
+        | ${count} =                 | Get Matching Locator Count | name:div_name                |
+        | Should Be Equal As Numbers | ${count}                   | 3                            |
+        | ${count} =                 | Get Matching Locator Count | div_name                     |
+        | Should Be Equal As Numbers | ${count}                   | 3                            |
```

Comment:
Do we need all these variations? This looks more like an example about using different locators. A reference to the locator section in the introduction ought to be enough.

Also, it's a bit funny to validate the count when the note above says that `Locator Should Match X Times` should be used for that instead. Perhaps this example could have something like

    Should Be True    ${count} > 3

and word "exact" be added to the above note like "verify the exact number of matching elements".

Finally, we should use proper English with `.` at the end of sentences. In this doc it's missing from the first and last sentence.

Reply:
The purpose of the example was to demonstrate that it's possible to get count by using different locators, but I agree that it was not good idea. It will only confuse users and is a maintenance burden.